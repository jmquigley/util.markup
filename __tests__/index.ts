import * as fs from "fs-extra";
import * as path from "path";
import assert from "power-assert";
import {cleanup, Fixture} from "util.fixture";
import {join} from "util.join";
import MarkupFactory, {HTMLResults, MarkupMode, MarkupTool} from "../index";

const debug = require("debug")("util.markup::test");

afterAll((done) => {
	cleanup({done, message: path.basename(__filename)});
});

test("Retrieve a asciidoc parser and convert to HTML", () => {
	const parser: MarkupTool = MarkupFactory.instance(MarkupMode.asciidoc);
	assert(parser);
	const fixture = new Fixture("asciidoc");
	assert(fixture);
	const txt = fixture.read("file.txt");
	assert(txt);
	const outfile = join(fixture.dir, "test.html");

	return parser
		.parse({markup: txt, outfile})
		.then((results: HTMLResults) => {
			expect(results.html).toMatchSnapshot();
			assert(fs.existsSync(results.filename));
		})
		.catch((err: string) => console.error(err));
});

test("Retrieve a markdown parser and convert to HTML", () => {
	const parser: MarkupTool = MarkupFactory.instance(MarkupMode.markdown);
	assert(parser);
	const fixture = new Fixture("markdown");
	assert(fixture);
	const md = fixture.read("file.md");
	assert(md);
	const outfile = join(fixture.dir, "test.html");

	return parser
		.parse({markup: md, outfile})
		.then((results: HTMLResults) => {
			expect(results.html).toMatchSnapshot();
			assert(fs.existsSync(results.filename));
		})
		.catch((err: string) => console.error(err));
});

test("Retrieve a restructuredtext parser and convert to HTML", () => {
	const parser: MarkupTool = MarkupFactory.instance(
		MarkupMode.restructuredtext
	);
	assert(parser);
	const fixture = new Fixture("restructuredtext");
	assert(fixture);
	const rst = fixture.read("file.rst");
	assert(rst);
	const outfile = join(fixture.dir, "test.html");

	return parser
		.parse({markup: rst, outfile})
		.then((results: HTMLResults) => {
			expect(results.html).toMatchSnapshot();
			assert(fs.existsSync(results.filename));
		})
		.catch((err: string) => console.error(err));
});
