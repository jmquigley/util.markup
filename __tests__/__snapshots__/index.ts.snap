// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Retrieve a asciidoc parser and convert to HTML 1`] = `
"
<!DOCTYPE html>

<html lang=\\"en\\">
	<head>
		<meta charset=\\"utf-8\\" />
		<meta name=\\"description\\" content=\\"util.markup\\" />
		<style>
		
		h2 {color: #ea1a1a};
	
		</style>
	</head>

	<body>
	<div class=\\"paragraph\\">
<p>some .</p>
</div>
<div class=\\"sect1\\">
<h2 id=\\"_test\\">test</h2>
<div class=\\"sectionbody\\">
<div class=\\"sect2\\">
<h3 id=\\"_test_2\\">test</h3>
<div class=\\"sect3\\">
<h4 id=\\"_test_3\\">test</h4>
<div class=\\"sect4\\">
<h5 id=\\"_test_4\\">test</h5>

</div>
</div>
</div>
</div>
</div>
<div class=\\"sect1\\">
<h2 id=\\"_test_5\\">test</h2>
<div class=\\"sectionbody\\">
<div class=\\"sect2\\">
<h3 id=\\"_test_6\\">test</h3>
<div class=\\"sect3\\">
<h4 id=\\"_test_7\\">test</h4>
<div class=\\"sect4\\">
<h5 id=\\"_test_8\\">test</h5>
<div class=\\"sect5\\">
<h6 id=\\"_test_9\\">test</h6>
<div class=\\"paragraph\\">
<p>this is <strong>bold</strong>
this is <em>italic</em>
this is mono or <code>mono</code>
this is <span class=\\"underline\\">underline</span>
this is <span class=\\"line-through\\">strikethrough</span></p>
</div>
<div class=\\"paragraph\\">
<p><span class=\\"red\\">this is a span of text</span></p>
</div>
<div class=\\"paragraph\\">
<p>[[block id element]]</p>
</div>
<div class=\\"paragraph\\">
<p>Symbols: &#174; &#169; (V) &#8482; &#8594; &#8592; &#8658; &#8656; &#182;&#8201;&#8212;&#8201;</p>
</div>
<div class=\\"dlist\\">
<dl>
<dt class=\\"hdlist1\\">ifdef</dt>
<dd>
<p>ifndef:: endif:: ifeval:: include:: include1:: sys:: eval:: template::</p>
</dd>
</dl>
</div>
<div class=\\"paragraph\\">
<p><a href=\\"http://www.google.com\\">Google</a> \\"this is some more text\\"
<a href=\\"https://www.example.com\\">Example</a>
<a href=\\"ftp://www.example.com\\">Example</a>
<a href=\\"file:///www.example.com\\">Example</a>
<a href=\\"http://www.example.com\\">Another Example</a></p>
</div>
<div class=\\"paragraph\\">
<p><a href=\\"http://www.example.com\\" class=\\"bare\\">http://www.example.com</a></p>
</div>
<div class=\\"paragraph\\">
<p><a href=\\"mailto:something@example.com\\">something@example.com</a>
<a href=\\"mailto:joe.bloggs@foobar.com\\">email Joe Bloggs</a></p>
</div>
<div class=\\"listingblock\\">
<div class=\\"title\\">Optional title</div>
<div class=\\"content\\">
<pre>let blah: number = \\"stuff\\";

function test() {
  if (test) {
    for (let i = 0; i &lt; 10; i++) {
      console.log('blah');
    }
  }
}</pre>
</div>
</div>
<div class=\\"listingblock\\">
<div class=\\"content\\">
<pre> More Code</pre>
</div>
</div>
<div class=\\"listingblock\\">
<div class=\\"content\\">
<pre>Code</pre>
</div>
</div>
<div class=\\"listingblock\\">
<div class=\\"content\\">
<pre>SQL&gt; describe aircraft;
 Name                   Null?  Type
 ----------------------------------------- -------- ----------------------------
 SERIAL_NO                 NOT NULL NUMBER(38)
 MODEL_NO                 NOT NULL VARCHAR2(15)

SQL&gt; describe flight;
 Name                   Null?  Type
 ----------------------------------------- -------- ----------------------------
 NUM                    NOT NULL NUMBER(38)
 ORIGIN                       VARCHAR2(3)
 DEST                        VARCHAR2(3)
 DEP_TIME                      VARCHAR2(5)
 ARR_TIME                      VARCHAR2(5)</pre>
</div>
</div>
<div class=\\"paragraph\\">
<p>between</p>
</div>
<div class=\\"literalblock\\">
<div class=\\"content\\">
<pre>// This is another line comment</pre>
</div>
</div>
<div class=\\"olist arabic\\">
<ol class=\\"arabic\\">
<li>
<p>slkdjfl</p>
<div class=\\"olist loweralpha\\">
<ol class=\\"loweralpha\\" type=\\"a\\">
<li>
<p>laskdjflkajsdlfkj</p>
<div class=\\"olist lowerroman\\">
<ol class=\\"lowerroman\\" type=\\"i\\">
<li>
<p>alskjdflaksjdf</p>
<div class=\\"olist upperalpha\\">
<ol class=\\"upperalpha\\" type=\\"A\\">
<li>
<p>asldkfjalksdjflkasjdf
.
..</p>
</li>
</ol>
</div>
</li>
<li>
<p>alsdjflaksjdflkajs</p>
</li>
</ol>
</div>
</li>
</ol>
</div>
</li>
</ol>
</div>
<div class=\\"paragraph\\">
<p>-
*</p>
</div>
<div class=\\"paragraph\\">
<p><strong>
</strong>*</p>
</div>
<div class=\\"paragraph\\">
<p>1.
2.
3.
4.</p>
</div>
<div class=\\"olist arabic\\">
<ol class=\\"arabic\\">
<li>
<p>alkjsdf lakjsd lfk</p>
</li>
<li>
<p>alskdjf laksdjf</p>
<div class=\\"ulist\\">
<ul>
<li>
<p>alksdj flasjd flakjds f</p>
</li>
</ul>
</div>
</li>
<li>
<p>laskjd flakjsd flkajd f</p>
</li>
<li>
<p>asldkgjalksdjgalksdgj</p>
<div class=\\"ulist\\">
<ul>
<li>
<p>slk <strong>djflsk</strong> jdf</p>
</li>
</ul>
</div>
</li>
<li>
<p>alksdjflkasjdflkaj sdlkfj</p>
<div class=\\"olist loweralpha\\">
<ol class=\\"loweralpha\\" type=\\"a\\">
<li>
<p>alskdjflkajsdlfkja</p>
</li>
<li>
<p>alksjdflkajsdlkfjalsdkfj</p>
</li>
</ol>
</div>
</li>
<li>
<p>alksdjflkasjdfkl</p>
</li>
<li>
<p>alskdj <em>flaksjdflkaj</em> sdfkljasldkfjlasdkfj</p>
<div class=\\"ulist\\">
<ul>
<li>
<p>laskdj flkasj dflkaj</p>
<div class=\\"ulist\\">
<ul>
<li>
<p>alskdjflaksdjf</p>
<div class=\\"ulist\\">
<ul>
<li>
<p>alksdjflaksdjfklajsdf</p>
</li>
<li>
<p>alksdjflaksdjf</p>
</li>
</ul>
</div>
</li>
<li>
<p>alksjdflkajsdfklj</p>
</li>
</ul>
</div>
</li>
<li>
<p>laksjdflaksdjflkajsdlfkjklajsdlfkjalsdkf</p>
</li>
</ul>
</div>
</li>
</ol>
</div>
<div class=\\"paragraph\\">
<p>TODO: lorem ipsum
FIXME: lorem ipsum
IMPORTANT: lorem ipsum
WARNING: lorem ipsum
TIP: lorem ipsum</p>
</div>
</div>
</div>
</div>
</div>
</div>
</div>
	</body>
</html>
"
`;

exports[`Retrieve a markdown parser and convert to HTML 1`] = `
"
<!DOCTYPE html>

<html lang=\\"en\\">
	<head>
		<meta charset=\\"utf-8\\" />
		<meta name=\\"description\\" content=\\"util.markup\\" />
		<style>
		
		h2 {color: #ea1a1a};
	
		</style>
	</head>

	<body>
	<h1>text</h1>
<h2>text</h2>
<h3>text</h3>
<h4>text</h4>
<h5>text</h5>
<h6>text</h6>
<h1>text</h1>
<p>lorem ipsum</p>
<h2>text</h2>
<p>lorem ipsum</p>
<p><em>italic</em>
<strong>bold</strong>
<code>monospace</code>
<em>underline</em>
~strikethrough~</p>
<hr>
<hr>
<hr>
<p>&lt;!-- comment block --&gt;</p>
<p>&lt;!--
comment block
--&gt;</p>
<ul>
<li>lorem ipsum 1</li>
<li>lorem ipsum 2</li>
</ul>
<ul>
<li>lorem ipsum 1</li>
<li>lorem ipsum 2</li>
</ul>
<ul>
<li>lorem ipsum 1</li>
<li>tlorem ipsum 2</li>
</ul>
<ol>
<li>lorem ipsum 1</li>
<li>lorem ipsum 2</li>
<li>lorem ipsum 3
a. lorem ipsum
b. lorem ipsum
c. lorem ipsum</li>
</ol>
<p>TODO: lorem ipsum
FIXME: lorem ipsum
IMPORTANT: lorem ipsum
WARNING: lorem ipsum
TIP: lorem ipsum</p>

	</body>
</html>
"
`;

exports[`Retrieve a restructuredtext parser and convert to HTML 1`] = `
"
<!DOCTYPE html>

<html lang=\\"en\\">
	<head>
		<meta charset=\\"utf-8\\" />
		<meta name=\\"description\\" content=\\"util.markup\\" />
		<style>
		
		h2 {color: #ea1a1a};
	
		</style>
	</head>

	<body>
	
<div>

<h1>text</h1>
lorem ipsum
<br /><br />

<h2>text</h2>
lorem ipsum
<br /><br />

<h3>text</h3>
lorem ipsum
<br /><br />

<h4>text</h4>
lorem ipsum
<br /><br />

<h5>test</h5>
lorem ipsum
<br /><br />

<h6>test</h6>
Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Tellus orci ac auctor augue mauris augue neque. Purus semper eget duis at tellus at. Et ultrices neque ornare aenean euismod elementum nisi quis eleifend. Vel orci porta non pulvinar. At auctor urna nunc id. Suscipit tellus mauris a diam maecenas sed. Sit amet nulla facilisi morbi tempus iaculis urna id volutpat. Neque volutpat ac tincidunt vitae semper quis lectus nulla at. Ultrices vitae auctor eu augue ut. Commodo elit at imperdiet dui. Elementum sagittis vitae et leo duis ut diam. Suscipit adipiscing bibendum est ultricies integer quis auctor elit. Neque convallis a cras semper auctor neque vitae. Ultrices neque ornare aenean euismod elementum nisi quis eleifend quam. Nunc aliquet bibendum enim facilisis gravida neque convallis a.
<br /><br />
Facilisis leo vel fringilla est ullamcorper. Elit at imperdiet dui accumsan sit amet. Ullamcorper eget nulla facilisi etiam dignissim diam. Arcu cursus vitae congue mauris rhoncus aenean vel elit scelerisque. Proin fermentum leo vel orci porta. Ullamcorper sit amet risus nullam. Blandit volutpat maecenas volutpat blandit aliquam etiam erat velit. Facilisis volutpat est velit egestas dui id ornare arcu. Amet est placerat in egestas erat imperdiet. In egestas erat imperdiet sed. Eget est lorem ipsum dolor sit amet consectetur adipiscing elit. Congue quisque egestas diam in arcu cursus euismod quis. Odio pellentesque diam volutpat commodo sed egestas. Non blandit massa enim nec.
<br /><br />
<em>italic</em>
<strong>bold</strong>
<code>monospace</code>
<br /><br />

<dl>
<dt>what</dt><dd>Definition lists associate a term with
a definition
<br /><br />
</dd><dt>how</dt><dd>the term is a one-line phrase, and the
definition is on eore more paragraphs or
body elements, indented relative to the term.
<br /><br />
</dd></dl>
<span class=\\"reference\\">reference</span>
<span class=\\"reference\\">phrase reference</span>
<span class=\\"reference\\">substitution reference</span>
<span class=\\"reference\\">1</span>
<span class=\\"reference\\">CIT2002</span>
.. [*] This is the first one.
.. _Python: http://www.python.org/
<span class=\\"reference\\">Python <http://www.python.org/></span>
.. image:: images/ball1.gif
.. <span class=\\"reference\\">biohazard</span> image:: biohazard.png
<br /><br />
http://www.example.com
test@example.com
<br /><br />

<code><pre>
let blah: number = \\"stuff\\";

function test() {
    if (test) {
        for (let i = 0; i < 10; i++) {
            console.log('blah');
        }
    }
}
</pre></code>
lorem ipsum :attribute: and a :keyword:
<br /><br />

<div class=\\"line_block\\">
<br />block&nbsp;text
<br />block&nbsp;text
<br />block&nbsp;text
</div>

<ul>
<li>lorem ipsum 1</li>
<li>lorem ipsum 2</li>
</ul>

<ul>
<li>lorem ipsum 1</li>
<li>lorem ipsum 2</li>
</ul>

<ul>
<li>lorem ipsum 1</li>
<li>tlorem ipsum 2</li>
</ul>

<ol>
<li>lorem ipsum 1</li>
<li>lorem ipsum 2</li>
</ol>

<ol>
<li>lorem ipsum 1</li>
<li>lorem ipsum 2</li>
<li>lorem ipsum 3
<ol>
<li>lorem ipsum</li>
<li>lorem ipsum</li>
<li>lorem ipsum</li>
</ol></li>
</ol>
TODO: lorem ipsum
FIXME: lorem ipsum
IMPORTANT: lorem ipsum
WARNING: lorem ipsum
TIP: lorem ipsum
<br /><br />






</div>

	</body>
</html>
"
`;

exports[`Retrieve an empty Markdown file 1`] = `""`;

exports[`Try to call parse with no options 1`] = `""`;
